flink-libraries/flink-table/src/main/scala/org/apache/flink/table/dataview/NullSerializer.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.dataview
+
+import org.apache.flink.api.common.typeutils.base.TypeSerializerSingleton
+import org.apache.flink.core.memory.{DataInputView, DataOutputView}
+
+class NullSerializer extends TypeSerializerSingleton[Any] {
```

Comment:
I am assuming `NullSerializer` is needed because the actual map is actually handled by StateBackend and because we removed them through `removeStateViewFieldsFromAccTypeInfo`. I was wondering why is this needed since it was not introduced in previous approach. Neither was it reflected in the test case ( still passes without the nullserializer setting)

Reply:
I think the reason `NullSerializer` is not needed previously is the magic of PojoTypeInfo(PojoTypeSerializer). PojoTypeInfo requires to specify the PojoFields for all the fields we want to serialize. For a DataView field, when we remove it through `removeStateViewFieldsFromAccTypeInfo`, we have also constructed a new PojoTypeInfo which removed the corresponding `PojoField` of the DataView field. While for TupleTypeInfo and CaseClassTypeInfo, they have no such magic.

Reply:
Yeah. I think I also trace the same code path for Pojo. However, removing the `NullSerializer` still lets the newly modified `testGroupAggregateWithStateBackend` passes. this is worrying because essentially there's no test safeguard for this piece of code being removed in the future.

Reply:
The tests can still pass without NullSerializer is because NullSerializer is an optimization. It doesn't affect the correctness. So an IT case can not cover this kind of tests. May be we should add a unit test for UserDefinedFunctionUtil#removeStateViewFieldsFromAccTypeInfo. But I found that currently there is no unit test for UserDefinedFunctionUtil. What do you think about adding a unit test class for UserDefinedFunctionUtil?

Reply:
hmm. I think that's a good idea but should probably be addressed in another JIRA. I actually have [PR:6472](https://github.com/apache/flink/pull/6472) still pending which I am also very fond of the idea to add test for the Utility class.

Reply:
What about only adding UserDefinedFunctionUtilTest#testRemoveStateViewFieldsFromAccTypeInfo in this PR? Just like you said, the tests of other methods of UserDefinedFunctionUtil can be done in another JIRA if needed.